### DISCUSSION TITLE
[PHP Imagick] How to set Bit Depth and Compression of file?

### DISCUSSION THREAD

**Discussion by jordsta95** [ImageMagick/ImageMagick#3605] [Help] [answered]
I am using Imagick to convert "standard" image formats, PNG/JP(E)G/BMP/etc., into a TGA, which is required for a 3rd party program.

However, the 3rd party requires that the TGA files have 32bpp, are uncompressed, and have no color palette.

I tried the following, however it didn't seem to change anything. Not too sure what the options need to be, or if there's an obvious setting I'm missing:

```
$imagick = new Imagick();
$imagick->readimageblob($file);
$imagick->setformat('tga');
$imagick->setOption('depth', '32');
$imagick->setOption('compress', 'None');
$tgaFile= $imagick->getImageBlob();
```

**Comment by fmw42** [comment, marked as answer]
depth 32 is not going to help. Set the type to truecolormatte. See https://www.php.net/manual/en/imagick.setimagetype.php

**Reply by jordsta95** [reply]
You're a genius! Thanks so much!

Wouldn't have even thought to look at that. Worked a charm 😄
